src/olympia/amo/tests/test_helpers.py

```diff
@@ -283,49 +284,97 @@ def test_external_url():
         settings.REDIRECT_SECRET_KEY = secretkey
 
 
-@patch('olympia.amo.templatetags.jinja_helpers.urlresolvers.get_outgoing_url')
-def test_linkify_bounce_url_callback(mock_get_outgoing_url):
-    mock_get_outgoing_url.return_value = 'bar'
-
-    res = urlresolvers.linkify_bounce_url_callback({(None, 'href'): 'foo'})
+def _test_linkify(func, value, expected_text, expected_links):
+    res = func(value)
+    # Make sure we didn't accidentally wrap things in a paragraph
+    assert not res.startswith('<p')
+    # Attributes could be rendered in any order so we can't test HTML directly.
+    # But PyQuery text() unescapes escaped HTML, so we double-check what we
+    # intended to escape too.
+    doc = PyQuery(res)
+    assert doc.text() == expected_text
+    # expected_text doesn't show escaped HTML, so we double-check we escaped
+    # things correctly.
+    assert '<script' not in res
 
-    # Make sure get_outgoing_url was called.
-    assert res == {(None, 'href'): 'bar'}
-    mock_get_outgoing_url.assert_called_with('foo')
+    for idx, expected_link in enumerate(expected_links):
+        link = doc('a')[idx]
+        assert link.attrib == {'href': expected_link[0], 'rel': 'nofollow'}
+        assert link.text == expected_link[1]
 
 
+@pytest.mark.parametrize(
+    'value,expected_text,expected_links',
+    [
+        (
+            'some http://foo.com text',
+            'some http://foo.com text',
+            (('https://out.going.com/http%3A//foo.com', 'http://foo.com'),),
+        ),
+        (
+            'some http://foo.com <b>bold</b>',
+            'some http://foo.com bold',
+            (('https://out.going.com/http%3A//foo.com', 'http://foo.com'),),
+        ),
+        (
+            'some http://foo.com <script>alert(42)</script> thing',
+            'some http://foo.com thing',
```

Comment:
Behavior change: if you tried to run that same test through the old implementation, it would escape `<script>alert(42)</script>`. `nh3` completely removes it.

Reply:
This is because of https://github.com/rust-ammonia/ammonia/issues/145 and I'm not super happy about it, might need to find a workaround, at the very least for markdown content.